### PR TITLE
feat: Port evaluator setup from dispatch-kit v0.4.1 (ADV-0051)

### PR DESCRIPTION
**Pull request by movito**
## Summary

Ports the evaluator setup from [dispatch-kit v0.4.1](https://github.com/movito/dispatch-kit/releases/tag/v0.4.1) into adversarial-workflow:

- **5 custom evaluators**: architecture-planner, architecture-planner-fast, architecture-reviewer, code-reviewer, spec-compliance
- **2 library evaluators**: arch-review-fast, code-reviewer-fast (Gemini Flash)
- **19 provider evaluators**: Anthropic (3), Google (6), OpenAI (10), Mistral (3)
- **EVALUATION-WORKFLOW.md**: Usage guide with fast/deep pairs, deprecation notice for built-in evaluators

Previously we only had the 3 built-in GPT-4o evaluators. Now `adversarial list-evaluators` shows 26 evaluators across 4 providers.

### Changes from PR #36

This replaces the closed PR #36 with files from dispatch-kit v0.4.1, which includes [fixes for all 14 issues](https://github.com/movito/dispatch-kit/issues/69) identified by bot reviews on the previous PR.

## Test plan

- [x] `adversarial list-evaluators` shows all 26 evaluators
- [x] Custom evaluator symlinks resolve correctly
- [x] dispatch-kit v0.4.1 fixes confirmed (gemini-deep→pro, mistral-fast updated, etc.)
- [ ] CI passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Low Risk**
> Primarily adds configuration/docs (YAML evaluators and a small symlink script) with no changes to core runtime logic; main risk is misconfigured evaluator YAML causing runtime selection/errors when invoked.
> 
> **Overview**
> Ports the evaluator setup from dispatch-kit into `.adversarial/`, adding a large set of named evaluators across OpenAI/Google/Anthropic/Mistral plus project-specific *fast/deep* pairs for planning, architecture review, code review, and spec compliance.
> 
> Adds `.adversarial/docs/EVALUATION-WORKFLOW.md` to document recommended usage, migration from deprecated built-in commands, required API keys, and log output locations. Includes a `custom/link-custom.sh` helper and top-level evaluator shim files (e.g. `architecture-planner.yml`) to expose custom evaluators via the CLI.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit ec72836f37043b2806a4506044992c28811624d6. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by movito** [comment]
Updated evaluators from dispatch-kit v0.4.2 (abd4f17). All 24 items from round 2 ([dispatch-kit#70](https://github.com/movito/dispatch-kit/issues/70)) are addressed:

- `gemini-pro` → `gemini-3.1-pro-preview` (urgent — old model deprecated tomorrow)
- `claude-adversarial` prompt injection guard added
- `code-reviewer-fast` non-code fallback added
- 20 minor fixes (formatting, consistency, docs)

Source: [dispatch-kit v0.4.2](https://github.com/movito/dispatch-kit/releases/tag/v0.4.2)

File: .adversarial/evaluators/google/gemini-code/evaluator.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+  ### Positive Observations
+  [Note security measures or patterns done well]
+
+  ## Overall Assessment
+
+  Provide a verdict:
+  - **APPROVED**: No critical/high issues, code is production-ready
+  - **CHANGES_REQUESTED**: Issues found that must be addressed
+  - **REJECT**: Critical security flaws requiring significant rework
```
[the remark this answers is not in the document]
Comment (movito) [reply .adversarial/evaluators/google/gemini-code/evaluator.yml]: Fixed in ec72836 — added untrusted-content guard matching the pattern from claude-adversarial.

File: .adversarial/evaluators/google/gemini-flash/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+Requires Google AI API key. Set in environment:
+```bash
+export GEMINI_API_KEY="your-key-here"
+```
+
+## Example Usage
+
+```bash
+adversarial evaluate evaluators/google/gemini-flash/evaluator.yml document.md
```
[the remark this answers is not in the document]
Comment (movito) [reply .adversarial/evaluators/google/gemini-flash/README.md]: Fixed in ec72836 — replaced `gpt52-reasoning` with `o3-chain or claude-adversarial`.

File: .adversarial/evaluators/openai/arch-review/README.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+```bash
+# Review a core module's architecture
+adversarial evaluate --evaluator arch-review src/dispatch_kit/bus.py
+
+# Review a data model's structure
+adversarial evaluate --evaluator arch-review src/models/event.py
+
+# Review an entire package
+adversarial evaluate --evaluator arch-review src/dispatch_kit/
```
[the remark this answers is not in the document]
Comment (movito) [reply .adversarial/evaluators/openai/arch-review/README.md]: Fixed in ec72836 — updated cross-references to use project evaluator names (`code-reviewer`, `code-reviewer-fast`) and corrected relative links.

File: .adversarial/evaluators/google/arch-review-fast/evaluator.yml [hunk 4]
```diff
@@ -0,0 +1,85 @@
+# Gemini Deep Think Architectural Review Evaluator
+# Extended reasoning analysis of code architecture — faster and cheaper alternative to arch-review
```
[the remark this answers is not in the document]
Comment (movito) [reply .adversarial/evaluators/google/arch-review-fast/evaluator.yml]: Dismissing — the description accurately lists what the evaluator checks for. "Design document conformance" means it checks if code matches stated patterns/contracts, which it does via the prompt. No separate design-doc parameter is needed.

File: .adversarial/evaluators/google/gemini-pro/evaluator.yml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+#   - Research completeness verification
+
+name: gemini-pro
+description: Large-context knowledge synthesis using Gemini 3 Pro (1M tokens)
+model: gemini/gemini-3-pro
+api_key_env: GEMINI_API_KEY
+model_requirement:
+  family: gemini
+  tier: pro
+  min_version: "3"
```
[the remark this answers is not in the document]
Comment (movito) [reply .adversarial/evaluators/google/gemini-pro/evaluator.yml]: Dismissing — same as above. The `gemini/` prefix is the CLI's model routing convention. The model value `gemini/gemini-3.1-pro-preview` is correct for this tool's configuration format.

File: .adversarial/evaluators/custom/code-reviewer.yml [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+# Deep correctness review — finds edge-case bugs that bots and Claude miss
+#
+# Use cases:
+#   - Pre-merge code review (after bot rounds)
+#   - Edge case and boundary condition analysis
+#   - Cross-model adversarial review (o1 vs Claude)
+#   - Resource exhaustion and timeout analysis
+
+name: code-reviewer
+description: Adversarial correctness review — finds bugs bots and Claude miss
```
[the remark this answers is not in the document]
Comment (movito) [reply .adversarial/evaluators/custom/code-reviewer.yml]: By design — custom evaluators take precedence over provider ones. The `custom/code-reviewer.yml` is the authoritative version; `openai/code-reviewer/` serves as a reference. The CLI already handles this correctly (skips duplicates with a warning).

File: .adversarial/docs/EVALUATION-WORKFLOW.md [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+> `adversarial proofread`, `adversarial review`) are **deprecated**. Use library
+> and custom evaluators instead — they support multiple providers (OpenAI, Google,
+> Anthropic, Mistral), configurable models, and role-specific evaluation.
+>
+> **Quick migration**:
+> - `adversarial evaluate <file>` -> `adversarial architecture-planner-fast <file>`
+> - `adversarial review <file>` -> `adversarial code-reviewer-fast <file>`
+> - For deep analysis: `adversarial architecture-planner <file>` or `adversarial code-reviewer <file>`
+>
+> Run `adversarial list-evaluators` to see all available evaluators.
```
[the remark this answers is not in the document]
Comment (movito) [reply .adversarial/docs/EVALUATION-WORKFLOW.md]: The deprecation notice was already clarified in a previous commit (882d389) to say "default built-in presets" rather than the invocation path. The evaluator READMEs reference `adversarial evaluate --evaluator ...` which is the named-evaluator path and is not deprecated.

File: .adversarial/evaluators/anthropic/claude-adversarial/README.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+## Cognitive Diversity Note
+
+This evaluator provides Anthropic's perspective on adversarial analysis. Use alongside OpenAI's gpt52-reasoning for maximum blind-spot coverage through multi-provider review.
+
+## Configuration
+
+```yaml
+api_key_env: ANTHROPIC_API_KEY
+model: claude-4-opus-20260115
+```
```
[the remark this answers is not in the document]
Comment (movito) [reply .adversarial/evaluators/anthropic/claude-adversarial/README.md]: Upstream concern — filed with adversarial-evaluator-library for source fix.

File: .adversarial/evaluators/custom/link-custom.sh [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+
+SCRIPT_DIR="$(cd "$(dirname "$0")" && pwd)"
+EVALUATORS_DIR="$(dirname "$SCRIPT_DIR")"
+
+cd "$EVALUATORS_DIR"
+shopt -s nullglob
+for f in custom/*.yml; do
+    ln -sf "$f" "$(basename "$f")"
+    echo "  Linked: $(basename "$f") -> $f"
+done
```
[the remark this answers is not in the document]
Comment (movito) [reply .adversarial/evaluators/custom/link-custom.sh]: Upstream concern — filed with adversarial-evaluator-library for source fix.

File: .adversarial/evaluators/custom/spec-compliance.yml [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+  {content}
+
+  ## Compliance Check Protocol
+
+  ### 1. Acceptance Criteria Audit
+  For EACH acceptance criterion in the task spec:
+  - Is it implemented? (YES / PARTIAL / NO)
+  - Where is it implemented? (file:function)
+  - Is it tested? (YES / NO)
+  - Evidence: [brief code reference]
```
[the remark this answers is not in the document]
Comment (movito) [reply .adversarial/evaluators/custom/spec-compliance.yml]: Upstream concern — filed with adversarial-evaluator-library for source fix.

File: .adversarial/evaluators/google-code-reviewer-fast.yml [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+  2. **Trace each edge case**: Mentally execute the function. Does it handle the case correctly?
+  3. **Check test coverage**: Is there a test for this edge case? Flag gaps.
+  4. **Check interactions**: Could function A's output break function B's assumptions?
+
+  ## Output Format
+
+  ### Findings
+
+  **[CORRECTNESS/ROBUSTNESS/TESTING]: [Title]**
+  - **Location**: `file.py:function_name`
```
[the remark this answers is not in the document]
Comment (movito) [reply .adversarial/evaluators/google-code-reviewer-fast.yml]: Upstream concern — filed with adversarial-evaluator-library for source fix.

File: .adversarial/evaluators/google/gemini-flash/CHANGELOG.md [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+
+### Added
+- Initial release
+- Fast document assessment prompt
+- Validated timeout setting (180s)
+- Tested on documents up to 15k tokens
+
+### Validated
+- GTX-0006: Response times 6-23 seconds on large documents
+- Cost: $0.0038 per evaluation (11k token document)
```
[the remark this answers is not in the document]
Comment (movito) [reply .adversarial/evaluators/google/gemini-flash/CHANGELOG.md]: Upstream concern — filed with adversarial-evaluator-library for source fix.

File: .adversarial/evaluators/mistral/mistral-fast/README.md [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+- High-volume processing with Mistral perspective
+
+## Performance
+
+| Metric | Value |
+|--------|-------|
+| Typical response time | 30-60 seconds |
+| Timeout setting | 300 seconds |
+| Cost | ~$0.005-0.02 per evaluation |
+
```
[the remark this answers is not in the document]
Comment (movito) [reply .adversarial/evaluators/mistral/mistral-fast/README.md]: Upstream concern — filed with adversarial-evaluator-library for source fix.
